core/aws-lsp-core/src/util/workspaceUtils.ts

```diff
@@ -31,6 +34,11 @@ export async function readDirectoryRecursively(
     const formatter = options?.customFormatCallback ?? formatListing
 
     while (queue.length > 0) {
+        if (token?.isCancellationRequested) {
+            features.logging.info('cancelled listing directory')
```

Comment:
this is meant to be a general utility to use elsewhere (currently not), so may be worth changing `listing directory` here to `readDirectoryRecursively` even though right now they are the same.